tests/unit/actions/settings/test_settings.py

```diff
@@ -0,0 +1,151 @@
+    """Describe the test"""
+    decoration: int
+    """The test decoration"""
+    is_default: bool
+    """Is the current value equal to the default"""
+
+    def __str__(self):
+        """Provide a string representation.
+
+        :returns: The string representation of self
+        """
+        return self.comment
+
+
+ColorMenuTestEntries = (
+    ColorMenuTestEntry(
+        comment="default/green",
+        color=Color.GREEN,
+        decoration=Decoration.NORMAL,
+        is_default=True,
+    ),
+    ColorMenuTestEntry(
+        comment="not default/yellow",
+        color=Color.YELLOW,
+        decoration=Decoration.NORMAL,
+        is_default=False,
+    ),
+)
+
+
+@pytest.mark.parametrize(argnames="data", argvalues=ColorMenuTestEntries, ids=str)
+def test_color_menu(data: ColorMenuTestEntry):
+    """Test color menu for a val set to the default.
+
+    :param data: A test entry
+    """
+    entry = {"is_default": data.is_default}
+    assert color_menu(0, "", entry) == (data.color, data.decoration)
+
+
+@dataclass
+class ContentHeadingEntry:
+    """A test for content headings."""
+
+    color: int
+    """The color for menu entry"""
+    comment: str
+    """Describes the test"""
+    current_value: str
+    """The current value"""
+    default: str
+    """The default value"""
+    is_default: Optional[bool] = None
+    """Indicate if the current is equal to the default."""
+    name: str = "test settings entry"
+    """The name of the settings entry"""
+
+    def __post_init__(self):
+        """Set is_default after initialization."""
+        self.is_default = self.current_value == self.default
+
+    @property
+    def heading(self):
+        """Create the expected heading for this content.
+
+        :returns: The expected heading
+        """
+        if self.is_default:
+            heading = CONTENT_HEADING_DEFAULT
+        else:
+            heading = CONTENT_HEADING_NOT_DEFAULT
+        return heading.format(**asdict(self))
+
+    def __str__(self):
+        """Provide a string representation.
+
+        :returns: The string representation of self
+        """
+        return self.comment
+
+
+ContentHeadingEntries = (
+    ContentHeadingEntry(
+        color=Color.GREEN,
+        comment="same",
+        current_value="navigator",
+        default="navigator",
+    ),
+    ContentHeadingEntry(
+        color=Color.YELLOW,
+        comment="different",
+        current_value="ansible",
+        default="navigator",
+    ),
+)
+
+
+@pytest.mark.parametrize(argnames="data", argvalues=ContentHeadingEntries, ids=str)
+def test_content_heading(data: ContentHeadingEntry):
+    """Test menu generation.
+
+    :param data: The test data
+    """
+    width = 100
+    heading = content_heading(obj=asdict(data), screen_w=width)
+    heading_lines = 1
+    line_parts = 1
+
+    assert len(heading) == heading_lines
+
+    first_line = heading[0]
+    assert len(first_line) == line_parts
+
+    first_line_part = heading[0][0]
```

Comment:
It seems like the variable names are misleading/mixed up here. How about
```suggestion
    heading_lines = content_heading(obj=asdict(data), screen_w=width)
    heading_line_number = 1
    columns_number = 1

    assert len(heading_lines) == heading_line_number

    first_line = heading_lines[0]
    assert len(first_line) == columns_number

    first_line_part = heading_lines[0][0]
```

Reply:
I see. The language should match.

Reply:
The main point here is that the variable names should correspond to data they supposedly contain.

Reply:
Which is exactly why they were named the way they were.

Good thing to keep in mind though!